### ISSUE TITLE
Serious security problems in encrypted chat rooms

### ISSUE THREAD

**Issue by Polve** [vector-im/element-meta#669]
### Steps to reproduce

A few of the protocol problems are outlined by this researcher in a tweet storm:
https://twitter.com/tqbf/status/1575259743278563329

### Outcome

Related informations are on this arstechnica article: https://arstechnica.com/information-technology/2022/09/matrix-patches-vulnerabilities-that-completely-subvert-e2ee-guarantees/

I open this ticket to understand if there are serious implications and to know if there are known mitigations we can use to limit the scope of the problems (if any)

### Operating system

all

### Will you send logs?

No

**Comment by Polve** [comment]
I would like to add the "security" label but it seems like I can't

**Comment by ShadowJonathan** [comment]
I've had to defend matrix over this tweet a few times last night, and what it comes down to is that this security researcher, while having good comments on the surface of matrix and a quick read-through of that paper, has no real idea of what safeguards are in place to prevent some seemingly-easy-to-spot security problems.

For example, in that tweet above, where they mention room participation being not encrypted, they blow up the problem as if it's a critical flaw, while matrix has mechanisms in place (TOFU on a user's cross-signing public key) that make sure that these are preventable and detectable. The fail-safe here is breaking the ability to do E2EE, not compromising it.

---

While I understand the frustration and fear behind this twitter thread, I do not see any actionable items on Element's part, as the paper already details implementation bugs (that have now been fixed and shipped).

For more fundamental flaws, you'd have to petition the [matrix.org specification repository](https://github.com/matrix-org/matrix-spec/), and not on a repository that works for one of many clients in matrix.

**Even then**, I want to ask you if you could detail exactly which parts of the protocol need to be revised from that paper, possibly even create multiple issues based on what the paper tells you. Being constructive in that fashion will get this concern much further than posting an issue with "look at this tweet, everything is bad, fix it", pardon my language.

---

But even then, from a personal perspective, I want to ask you if you think this paper hasn't been looked at closely by the matrix.org foundation guardians and security researchers already, and to draw conclusions on their end. I think that highlighting this tweet back to them, with surface-level observations, on a paper that they have already gone through and fixed issues for, is not going to create a huge revelation that upends E2EE.

**Comment by Polve** [comment]
> For more fundamental flaws, you'd have to petition the [matrix.org specification repository](https://github.com/matrix-org/matrix-spec/), and not on a repository that works for one of many clients in matrix

You are right, I didn't think about it. I will, if it still makes sense to do it.

>Even then, I want to ask you if you could detail exactly which parts of the protocol need to be revised from that paper

I'm not technical enough to do it: my goal was just to raise the issue in order to have some insight from the matrix part since I didn't find any useful reply on that thread (or maybe I overlooked something).

> than posting an issue with "look at this tweet, everything is bad, fix it", pardon my language

No need to pardon, the problem is mine: I was unable to read a rebuttal view and I was afraid that the problem was overlooked so I thought I could have some more information here.

I apologize if I used the wrong wording or I didn't follow the correct protocol for this kind of issue, I just wanted to raise the awareness and verify that those kind of scenario was well understood.

Thanks for your reply, I close the issue but I will welcome any more useful pointers to discussions (articles or even twitter replies) that can shed more light on the subject, so feel free to add more info to the ticket if you are willing to.
